llvm/lib/Target/AIE/AIE2TargetMachine.cpp

```diff
@@ -139,6 +139,10 @@ void AIE2PassConfig::addIRPasses() {
           }));
     }
   }
+
+  if (TM->getOptLevel() > CodeGenOptLevel::None)
+    addPass(createInferAddressSpacesPass());
```

Comment:
I wouldn't mind always adding this. -O0 is not a useful optimization level anyway, since we always run out of stack.